src/topology/connected.lean

```diff
@@ -353,6 +353,79 @@ begin
   exact hc.image _ (continuous_apply _).continuous_on
 end
 
+lemma sigma.is_connected_iff [Π i, topological_space (π i)] {s : set (Σ i, π i)} :
+  is_connected s ↔ ∃ i t, is_connected t ∧ s = sigma.mk i '' t :=
+begin
+  refine ⟨λ hs, _, _⟩,
+  { obtain ⟨⟨i, x⟩, hx⟩ := hs.nonempty,
+    have : s ⊆ set.range (sigma.mk i),
+    { have h : set.range (sigma.mk i) = sigma.fst ⁻¹' {i}, by { ext, simp },
+      rw h,
+      exact is_preconnected.subset_left_of_subset_union
+        (is_open_sigma_fst_preimage _) (is_open_sigma_fst_preimage {x | x ≠ i})
+        (set.disjoint_iff.2 $ λ x hx, hx.2 hx.1)
+        (λ y hy, by simp [classical.em]) ⟨⟨i, x⟩, hx, rfl⟩ hs.2 },
+    exact ⟨i, sigma.mk i ⁻¹' s,
+      hs.preimage_of_open_map sigma_mk_injective is_open_map_sigma_mk this,
+      (set.image_preimage_eq_of_subset this).symm⟩ },
+  { rintro ⟨i, t, ht, rfl⟩,
+    exact ht.image _ continuous_sigma_mk.continuous_on }
+end
+
+lemma sigma.is_preconnected_iff [hι : nonempty ι] [Π i, topological_space (π i)]
+  {s : set (Σ i, π i)} :
+  is_preconnected s ↔ ∃ i t, is_preconnected t ∧ s = sigma.mk i '' t :=
+begin
+  refine ⟨λ hs, _, _⟩,
+  { obtain rfl | h := s.eq_empty_or_nonempty,
+    { exact ⟨classical.choice hι, ∅, is_preconnected_empty, (set.image_empty _).symm⟩ },
+    { obtain ⟨a, t, ht, rfl⟩ := sigma.is_connected_iff.1 ⟨h, hs⟩,
+      refine ⟨a, t, ht.is_preconnected, rfl⟩ } },
+  { rintro ⟨a, t, ht, rfl⟩,
+    exact ht.image _ continuous_sigma_mk.continuous_on }
+end
+
+lemma sum.is_connected_iff [topological_space β] {s : set (α ⊕ β)} :
+  is_connected s ↔
+    (∃ t, is_connected t ∧ s = sum.inl '' t) ∨ ∃ t, is_connected t ∧ s = sum.inr '' t :=
```

Comment:
Why not
```suggestion
    ∃ t, is_connected t ∧ (s = sum.inl '' t ∨ s = sum.inr '' t) :=
```

Reply:
I don't know which way is preferred, so I'll switch if you say so.